Experimental/src/org/sleuthkit/autopsy/experimental/autoingest/AutoIngestManager.java

```diff
@@ -2456,6 +2447,7 @@ private Case openCase() throws CoordinationServiceException, CaseManagementExcep
                             Thread.sleep(AutoIngestUserPreferences.getSecondsToSleepBetweenCases() * 1000);
                         }
                         currentJob.setCaseDirectoryPath(caseDirectoryPath);
+                        updateCoordinationServiceManifestNode(currentJob);  // update case directory path
```

Comment:
Other places that call updateCoordinationServiceManifestNode() get an exclusive lock on the manifest file node. This block of code has a lock on the case node.

Reply:
Originally I thought so too but turns out that's not the case. For example, PeriodicJobStatusEventTask.run() calls updateCoordinationServiceManifestNode() without obtaining the lock. Same goes for processJob(), which actually calls the openCase() method (the one I modified). So I think obtaining the lock seems only necessary when creating a new ZK node, as a mechanism for actual node creation, or perhaps for the purposes of not creating duplicate ZK nodes?

Reply:
Okay, I give up trying to wrap my head around it then.